src/OpenFeature.Contrib.Providers.Flipt/FliptProvider.cs

```diff
@@ -0,0 +1,155 @@
+﻿using Grpc.Core;
+using Grpc.Net.Client;
+using OpenFeature.Error;
+using OpenFeature.Model;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using static Flipt.Evaluation.EvaluationService;
+using EvaluationRequest = Flipt.Evaluation.EvaluationRequest;
+using Metadata = OpenFeature.Model.Metadata;
+
+namespace OpenFeature.Contrib.Providers.Flipt
+{
+    /// <summary>
+    /// Flipt feature provider.
+    /// </summary>
+    public sealed class FliptProvider : FeatureProvider
+    {
+        private static readonly Metadata Metadata = new Metadata("Flipt");
+        private readonly EvaluationServiceClient _client;
+        private readonly FliptProviderConfiguration _configuration;
+
+        /// <summary>
+        /// Creates new instance of <see cref="FliptProvider" />
+        /// </summary>
+        /// <param name="configuration">Flipt provider options.</param>
+        public FliptProvider(FliptProviderConfiguration configuration)
+        {
+            _configuration = configuration;
+            var channel = GrpcChannel.ForAddress(configuration.ServiceUri);
+            _client = new EvaluationServiceClient(channel);
```

Comment:
we may want to think about making a HTTP client and making that the default as GRPC requires HTTP2 and might not work in all environments.

I realize making an HTTP C# client for Flipt is not a small undertaking though, we've had this issue open for a bit in our server-side SDK monorepo: https://github.com/flipt-io/flipt-server-sdks/issues/86

Reply:
That's true. It would be great to use a client interface instead of the implementation itself (GRPC client SDK, HTTP client SDK or server SDK), but the library doesn't support that at the moment.
For now I can I suggest either modifying the repository [flipt-grpc-dotnet](https://github.com/flipt-io/flipt-grpc-dotnet) by adding a GRPC client interface, or making [this constructor](https://github.com/open-feature/dotnet-sdk-contrib/blob/81607213b7084e27682273cb3c8d5032036812dc/src/OpenFeature.Contrib.Providers.Flipt/FliptProvider.cs#L39) public

Reply:
Yeah we actually do something similar in our Go Open Feature provider, which supports both GRPC and HTTP

There its all based on the protocol in the address / URL string for which transport to use: https://github.com/open-feature/go-sdk-contrib/tree/main/providers/flipt#configuration

https://github.com/open-feature/go-sdk-contrib/blob/main/providers/flipt/pkg/service/transport/service.go#L152

We could do something similar here, where at first we would only support the GRPC transport like you have done, but once we ideally have a C# REST/HTTP SDK then we can add support here in a non-breaking way, by checking the protocol.

Reply:
I think this still needs addressing @dmitryrogov